### ISSUE TITLE
compile error

### ISSUE THREAD

**Issue by liaosiwei** [miloyip/rapidjson#433]
hi, I am using rapidjson to construct and modify json data in an existing project. First, I have written some demo with the rapidjson include files and compile them through. And later, I try to use the rapidjson in the existing project. I didn't change any code but just include one header file: <strong>document.h</strong> in one header file from the project and compile the project, then, error ocurrs. here is the error report:
<pre><code>In file included from rapidjson/document.h:20,
                 from idebug_ex.h:22,
                 from idebugcontrol.h:34,
                 from idebug_ex.cpp:19:
rapidjson/reader.h:190: error: expected unqualified-id before "unsigned"
rapidjson/reader.h:190: error: expected `)' before "unsigned"
rapidjson/reader.h: In member function `void idebug_rapidjson::GenericReader<SourceEncoding, TargetEncoding, StackAllocator>::ParseNumber(InputStream&, Handler&)':
rapidjson/reader.h:978: error: expected unqualified-id before '(' token
rapidjson/reader.h:978: error: expected primary-expression before "unsigned"
In file included from idebug_ex.h:22,
                 from idebugcontrol.h:34,
                 from idebug_ex.cpp:19:
rapidjson/document.h: In member function `bool idebug_rapidjson::GenericValue<Encoding, Allocator>::Accept(Handler&) const':
rapidjson/document.h:1568: error: expected unqualified-id before '(' token
rapidjson/document.h:1568: error: expected primary-expression before "unsigned"
rapidjson/document.h: At global scope:
rapidjson/document.h:2016: error: expected unqualified-id before "unsigned"
rapidjson/document.h:2016: error: expected `)' before "unsigned" </code></pre>
these messages seems like the library has some errors, but I know it's not true. But under this condition, I can't trace the bug source. Did any one have the same problem as mine, or anyone has any idea solving this problem?

I am using gcc 3.4.5 under redhat linux os

**Comment by mloskot** [comment]
> I am using gcc 3.4.5

The [compatibility](http://rapidjson.org/) section lists GCC 3.8. Possibly, 3.4 line is not supported.

**Comment by pah** [comment]
The compiler errors smell like a conflicting macro definition.  The first two lines with errors are `reader.h:190` and `reader.h:978`, both containing the `Uint` identifier.

Make sure to avoid defining a macro named `Uint` before including the RapidJSON headers.  Maybe a `typedef` is a better solution here?

That said, there might indeed be problems with GCC earlier than 3.8, but this needs to be seen after the macro conflict has been resolved.  You could try to run the regression tests on your platform first (and please report your findings here).

**Comment by liaosiwei** [comment]
After a simple search in the other dependent libraries, I found a defined macro which contains the Uint identifier. pah is right and I did this trick to prevent the macro conflictions:
<pre><code>
4   #ifdef Uint
5   #undef Uint
6   #include "rapidjson/document.h"
7   #include "rapidjson/prettywriter.h"
8   #include "rapidjson/writer.h"
9   #include "rapidjson/stringbuffer.h"
10  #define Uint XXXX // the macro definition in dependent library
11  #endif</code></pre>
then, I used the rapidjson lib to build the project with gcc 3.4.5, and the program runs normally. I didn't do the regression test in the source code. But it seems fine working with gcc 3.4.5.
thank you, all guys.
